fms_extras/models/paged_llama.py

```diff
@@ -0,0 +1,844 @@
+import json
+import math
+import os
+import re
+from collections import OrderedDict
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Mapping, MutableMapping, Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from fms import distributed, models
+from fms.distributed.strategy import (
+    DistributedStrategy,
+    NoOpStrategy,
+    UniformModelParallelStrategy,
+)
+from fms.distributed.tensorparallel import (
+    copy_to_tensor_model_parallel_region,
+    reduce_from_tensor_model_parallel_region,
+)
+from fms.modules.attention import MultiHeadAttention
+from fms.modules.embedding import WordEmbedding
+from fms.modules.feedforward import GatedLinearUnit
+from fms.modules.layernorm import LayerNormParameterized
+from fms.modules.positions import PositionEncoder, RotaryEmbedding
+from fms.modules.tp import TPModule
+from fms.utils import serialization
+from fms.utils.activation import str_to_activation
+from fms.utils.config import ModelConfig
+from torch._C._distributed_c10d import ProcessGroup
+
+from fms_extras.utils.cache.paged import (
+    PagedAttentionCacheData,
+    PagedAttentionCacheDataLayer,
+)
+
+
+# params emb_dim heads layers lr
+#  7B    4096    32    32     3.0E-04
+# 13B    5120    40    40     3.0E-04
+# 33B    6656    52    60     1.5.E-04
+# 65B    8192    64    80     1.5.E-04
+
+
+@dataclass
+class PagedLLaMAConfig(ModelConfig):
+    src_vocab_size: int = 32_000  # can be set by tokenizer
```

Comment:
I think the comment isn't clear here, a don't think using a tokenizer will lead to this getting updated automatically right?